modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperiments.es.js

```diff
@@ -68,7 +68,13 @@ function SegmentsExperiments({
 		: selectedExperienceId;
 	const noExperimentIllustration = `${imagesPath}${NO_EXPERIMENT_ILLUSTRATION_FILE_NAME}`;
 	const winnerVariant = variants.find((variant) => variant.winner === true);
-	const goalTarget = experiment?.goal?.target?.substring(1);
+	const goalTarget = experiment?.goal?.target?.replace('#', '');
+
+	// If the target has been removed from the page we must reset it
+
+	if (goalTarget && !document.getElementById(goalTarget)) {
```

Comment:
Same as before, let me know what you think => Can we condense this condition in a variable with a semantic name? If not, the comment is fine for me 👍